index.js

```diff
@@ -1,14 +1,264 @@
 const express = require('express');
 const connectDB = require('./config/db');
+const config = require('config');
+const nanoid = require('nanoid');
+
+
+const Url = require('./models/Url');
+const { listen } = require('express/lib/application');
 
 const app = express();
 
+
 // Connect to database
 connectDB();
 
 app.use(express.json());
 
-app.use('/', require('./routes/index'));
+// @route     POST /api/v1/shorten
+// @desc      Create short URL
+
+app.post('/api/v1/shorten', async (req, res) => {
+  try {
+    const { url, customName } = req.body;
+
+    // Validate longUrl
+    if (!url) {
+      return res.status(400).json({ error: 'Please provide a valid long URL.' });
+    }
+
+    let existingURL;
+
+    // Check if longUrl already exists
+    existingURL = await Url.findOne({ originalUrl: url });
+    if (existingURL) {
+      return res.json({
+        message: 'URL already exists',
+        data: {
+          shortUrl: existingURL.shortUrl,
+        },
+      });
+    }
+
+    // Validate customName if provided
+    if (customName) {
+      if (customName.length < 5) {
+        return res.status(400).json({ error: 'Custom name must be at least 5 characters long.' });
+      }
+
+      const formattedCustomName = customName.replace(/\s+/g, '-'); // Replace spaces with dashes
+      existingURL = await Url.findOne({ customName: formattedCustomName });
+
+      if (existingURL) {
+        return res.status(400).json({ error: 'Custom name already exists.' });
+      }
+
+      const newShortenedURL = new ShortenedURL({
+        customName: formattedCustomName,
+        shortUrl: `https://shortit/${formattedCustomName}`,
+        originalUrl: url,
+      });
+
+      await newShortenedURL.save();
+
+      return res.json({
+        message: 'URL shortened successfully',
+        data: {
+          shortUrl: newShortenedURL.shortUrl,
+        },
+      });
+    }
+
+    // Generate random shortUrl
+    const randomShortUrl = nanoid(5);
+    existingURL = await Url.findOne({ shortUrl: randomShortUrl });
+
+    // Ensure random shortUrl is unique
+    while (existingURL) {
+      randomShortUrl = nanoid(5);
+      existingURL = await Url.findOne({ shortUrl: randomShortUrl });
+    }
+
+    const newShortenedURL = new ShortenedURL({
+      shortUrl: `https://shortit/${randomShortUrl}`,
+      originalUrl: url,
+    });
+
+    await newShortenedURL.save();
+
+    return res.json({
+      message: 'URL shortened successfully',
+      data: {
+        shortUrl: newShortenedURL.shortUrl,
+      },
+    });
```

Comment:
You are repeating the same block of code here like the one [above](https://github.com/Godwin2222/Url-shortner-API/blob/f38b264d5dcb3d0c8c63f89b98dd0fd2ae72cd27/index.js#L56-L69). Your code must be DRY. Please refactor.